.github/workflows/deploy-cognito.yml

```
@@ -0,0 +1,61 @@
+name: Deploy Cognito stack to devplatform secure pipeline
+
+on:
+  push:
+    branches:
+      - main
+      - SSE-2450_cognito_deploying_to_build
+    paths:
+      - backend/cognito/**
+      - .github/workflows/**
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+
+  deploy:
+    name: Deploy Cognito
+    runs-on: ubuntu-latest
+    environment: development
+
+    steps:
+      - name: Pull repository
+        uses: actions/checkout@v3
+
+      - name: Install Node
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Set up SAM cli
+        uses: aws-actions/setup-sam@v2
+
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          role-to-assume: ${{ secrets.SSE2450_GHA_ROLE_ARN }}
+          aws-region: eu-west-2
+
+      - name: SAM validate
+        working-directory: backend/cognito
+        run: sam validate --template template.yml
```

```suggestion
        run: |
          sam validate --template template.yml
          sam validate --template template.yml --lint
```
Add more validation checks